src/SFA.DAS.EmployerIncentives.Commands/Services/LearnerMatchApi/LearnerDataExtensions.cs

```diff
@@ -171,8 +171,15 @@ private static IList<LearningPeriod> MergeLearningPeriods(IList<LearningPeriod>
         {
             for (var i = 0; i < periods.Count-1; i++)
             {
-                var gap = (periods[i + 1].StartDate - periods[i].EndDate).Days;
-                if (gap <= 1)
+                // merge learning periods if:
+                // periods are adjacent
+                // periods overlap
```

Comment:
Can you refactor the code so that they read as per the comments  e.g. if(PeriodsAreAdjacent() || PeriodsOverlap() ... etc...